components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2634,63 +2834,88 @@ private String getSearchAttribute(String filterOperation, String attributeValue,
-            throws org.wso2.carbon.user.core.UserStoreException, IdentitySCIMException {
+    private String[] getGroupList(ExpressionNode expressionNode, String domainName)
+            throws org.wso2.carbon.user.core.UserStoreException, CharonException {
+
+        String attributeName = expressionNode.getAttributeValue();
+        String filterOperation = expressionNode.getOperation();
+        String attributeValue = expressionNode.getValue();
 
-        String[] userRoleList;
-        if (attributeName.equals(SCIMConstants.GroupSchemaConstants.DISPLAY_URI)
-                || attributeName.equals(SCIMConstants.GroupSchemaConstants.VALUE_URI)) {
+        // Groups endpoint only support display uri and value uri.
+        if (attributeName.equals(SCIMConstants.GroupSchemaConstants.DISPLAY_URI) || attributeName
+                .equals(SCIMConstants.GroupSchemaConstants.VALUE_URI)) {
             String[] userList;
+
+            // Update attribute value with the domain name.
+            attributeValue = prependDomainNameToTheAttributeValue(attributeValue, domainName);
+            // Listing users.
             if (attributeName.equals(SCIMConstants.GroupSchemaConstants.DISPLAY_URI)) {
-                userList = getUserNames(SCIMConstants.UserSchemaConstants.USER_NAME_URI, filterOperation, attributeValue);
+                userList = getUserNames(SCIMConstants.UserSchemaConstants.USER_NAME_URI, filterOperation,
+                        attributeValue);
             } else {
                 userList = getUserNames(SCIMConstants.CommonSchemaConstants.ID_URI, filterOperation, attributeValue);
             }
+            // Get the roles of the users.
             Set<String> fullRoleList = new HashSet<>();
-            List<String> currentRoleList;
-
             if (userList != null) {
                 for (String userName : userList) {
-                    String[] roles = carbonUM.getRoleListOfUser(userName);
-                    currentRoleList = Arrays.asList(roles);
-                    fullRoleList.addAll(currentRoleList);
+                    fullRoleList.addAll(Arrays.asList(carbonUM.getRoleListOfUser(userName)));
                 }
             }
-
-            userRoleList = fullRoleList.toArray(new String[fullRoleList.size()]);
-
+            return fullRoleList.toArray(new String[0]);
         } else if (attributeName.equals(SCIMConstants.GroupSchemaConstants.DISPLAY_NAME_URI)) {
-            userRoleList = getRoleNames(filterOperation, attributeValue);
+            attributeValue = prependDomainNameToTheAttributeValue(attributeValue, domainName);
+            return getRoleNames(filterOperation, attributeValue);
         } else {
-            userRoleList = getGroupNamesFromDB(attributeName, filterOperation, attributeValue);
+            try {
+                return getGroupNamesFromDB(attributeName, filterOperation, attributeValue, domainName);
```

Comment:
Are we returning group names for attributes other than `display, display_name, user_name`?

Reply:
Yes, Attributes such as Meta etc.